framework/src/utils/lib/s3-data-copy.ts

```diff
@@ -0,0 +1,120 @@
+    };
+    super(scope, id, trackedConstructProps);
+
+    if (props.vpc === undefined && props.subnets !==undefined) {
+      throw new Error('S3DataCopy error: if VPC parameter is not configured, subnets cannot be');
+    }
+    if (props.vpc !== undefined && props.subnets === undefined) {
+      throw new Error('S3DataCopy error: if VPC parameter is configured, subnets must be');
+    }
+
+    const functionName = `s3-data-copy-${Utils.generateScopeHash(this)}`;
+
+    const managedPolicy = new ManagedPolicy(this, 'Policy', {
+      document: new PolicyDocument({
+        statements: [
+          new PolicyStatement({
+            effect: Effect.ALLOW,
+            actions: [
+              'logs:CreateLogGroup',
+              'logs:CreateLogStream',
+              'logs:PutLogEvents',
+            ],
+            resources: [
+              `arn:aws:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/lambda/${functionName}`,
+              `arn:aws:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/lambda/${functionName}:log-stream:*`,
+            ],
+          }),
+          new PolicyStatement({
+            actions: [
+              'ec2:CreateNetworkInterface',
+              'ec2:DescribeNetworkInterfaces',
+              'ec2:DeleteNetworkInterface',
+              'ec2:AssignPrivateIpAddresses',
+              'ec2:UnassignPrivateIpAddresses',
+            ],
+            effect: Effect.ALLOW,
+            resources: ['*'],
+          }),
+        ],
+      }),
+    });
+
+    this.copyRole = props.copyRole || new Role(this, 'Role', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      description: 'Role used by S3DataCopy to copy data from one S3 bucket to another',
+    });
+    this.copyRole.addManagedPolicy(managedPolicy);
+
+    props.sourceBucket.grantRead(this.copyRole, `${props.sourceBucketPrefix || ''}*`);
+    props.targetBucket.grantWrite(this.copyRole, `${props.targetBucketPrefix || ''}*`);
+
+    this.copyLambda = new NodejsFunction(this, 'Lambda', {
+      runtime: S3DataCopy.CR_RUNTIME,
+      role: this.copyRole,
+      entry: path.join(__dirname, './resources/lambda/s3-data-copy/index.mjs'),
+      timeout: Duration.minutes(15),
+      environment: {
+        SOURCE_BUCKET_NAME: props.sourceBucket.bucketName,
+        SOURCE_BUCKET_PREFIX: props.sourceBucketPrefix || '',
+        SOURCE_BUCKET_REGION: props.sourceBucketRegion,
+        TARGET_BUCKET_NAME: props.targetBucket.bucketName,
+        TARGET_BUCKET_PREFIX: props.targetBucketPrefix || '',
+      },
+      vpc: props.vpc,
+      vpcSubnets: props.vpc ? { subnets: props.subnets } : undefined,
+      logRetention: S3DataCopy.LOG_RETENTION,
+    });
+    // Custom resource provider
+    const copyProvider = new Provider(this, 'Provider', {
+      onEventHandler: this.copyLambda,
+      logRetention: S3DataCopy.LOG_RETENTION,
+    });
+
+    // Custom resource to trigger copy
+    new CustomResource(this, 'CustomResource', {
```

Comment:
can you add the parameter called `resourceType` and provide a value as follow 'DSF::S3DataCopy::Copy'. This is for visibility on the custom resource in the CFN console